addons/event_booth_sale/models/sale_order_line.py

```diff
@@ -38,11 +38,21 @@ def _compute_event_booth_pending_ids(self):
 
     def _inverse_event_booth_pending_ids(self):
         for so_line in self:
+            existing_booths = so_line.event_booth_registration_ids.event_booth_id
+            selected_booths = so_line.event_booth_pending_ids
+
+            booth_regs_to_remove = so_line.event_booth_registration_ids.filtered(
+                lambda reg: reg.event_booth_id not in selected_booths)
+            booth_regs_to_remove.unlink()
+
+            booths_to_create = selected_booths.filtered(
+                lambda booth: booth not in existing_booths) if existing_booths else selected_booths
+
             self.env['event.booth.registration'].create([{
                 'event_booth_id': booth.id,
                 'sale_order_line_id': so_line.id,
                 'partner_id': so_line.order_id.partner_id.id
-            } for booth in so_line.event_booth_pending_ids])
+            } for booth in booths_to_create])
```

Comment:
Can't you simplify to `for booth in selected_booths  - existing_booths` ?

You might have to slightly change the above code as well to:
`existing_booths = so_line.event_booth_registration_ids.event_booth_id or self.env['event.booth']` so that it subtracts an empty recordset if the variable is False.